Tools-Override/CodeCoverage.targets

```diff
@@ -0,0 +1,132 @@
+    <CoverageCommandLine>$(CoverageOptions) -returntargetcode -register:user -target:$(TestProgram) -output:$(CoverageOutputFilePath)</CoverageCommandLine>
+    <TestHost>$(CoverageHost)</TestHost>
+    <XunitOptions>$(XunitOptions) -parallel none</XunitOptions>
+    <TestCommandLine>$(TestHost) $(CoverageCommandLine) -targetargs:"$(TestArguments) {XunitTraitOptions}"</TestCommandLine>
+  </PropertyGroup>
+
+  <!-- Report Generator Properties -->
+  <PropertyGroup>
+    <CoverageReportAssemblyFilters>@(_CodeCoverageAssemblies->'+%(Identity)', ';')</CoverageReportAssemblyFilters>
+    <CoverageReportGeneratorOptions>-targetdir:$(CoverageReportDir) -reporttypes:Html;Badges "-assemblyfilters:$(CoverageReportAssemblyFilters)"</CoverageReportGeneratorOptions>
+    <CoverageReportGeneratorCommandLine>$(PackagesDir)ReportGenerator\$(ReportGeneratorVersion)\tools\ReportGenerator.exe $(CoverageReportGeneratorOptions)</CoverageReportGeneratorCommandLine>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(GenerateIndividualCoverageReport)'=='true'">
+    <GenerateIndividualCoverageReportAfterTargets Condition="'$(GenerateIndividualCoverageReportAfterTargets)'==''">RunTestsForProject</GenerateIndividualCoverageReportAfterTargets>
+  </PropertyGroup>
+
+  <!-- Generate coverage reports for individual projects. -->
+  <Target Name="GenerateIndividualCoverageReport"
+          AfterTargets="$(GenerateIndividualCoverageReportAfterTargets)"
+          Inputs="$(CoverageOutputFilePath)"
+          Outputs="$(CoverageReportDir)index.htm"
+          Condition="'$(GenerateIndividualCoverageReport)'=='true'">
+
+    <Exec Command="$(CoverageReportGeneratorCommandLine) -reports:$(CoverageOutputFilePath)"
+          ContinueOnError="ErrorAndContinue" />
+
+    <Exec Command="start $(CoverageReportDir)index.htm"
+          Condition="'$(PopCoverageReport)' == 'true'" />
+
+  </Target>
+
+  <PropertyGroup Condition="'$(GenerateFullCoverageReport)'=='true'">
+    <GenerateFullCoverageReportAfterTargets Condition="'$(GenerateFullCoverageReportAfterTargets)'==''">TestAllProjects</GenerateFullCoverageReportAfterTargets>
+  </PropertyGroup>
+
+  <!-- Generate coverage report for all the projects. -->
+  <Target Name="GenerateFullCoverageReport"
+         AfterTargets="$(GenerateFullCoverageReportAfterTargets)"
+         Inputs="$(CoverageReportDir)\*.coverage.xml"
+         Outputs="$(CoverageReportDir)index.htm"
+         Condition="'$(GenerateFullCoverageReport)'=='true'">
+
+    <Exec Command="$(CoverageReportGeneratorCommandLine) -reports:$(CoverageReportDir)\*.coverage.xml"
+          ContinueOnError="ErrorAndContinue"
+          WorkingDirectory="$(ProjectDir)" />
+
+    <Exec Command="start $(CoverageReportDir)index.htm"
+          Condition="'$(PopCoverageReport)' == 'true'" />          
+
+    <PropertyGroup>
+      <CoverallsUploaderCommandLine>$(PackagesDir)coveralls.io.$(CoverallsUploaderVersion)\tools\coveralls.net.exe</CoverallsUploaderCommandLine>
```

Comment:
@stephentoub No, but corert wants to.  @dleeapho This would be a good candidate to move into buildtools so that corert and corefx could share functionality.